src/utils.js

```diff
@@ -67,3 +69,37 @@ export function forNestedRelations(model, nestedRelations, fn) {
 
     });
 }
+
+// stale request utils
+
+/**
+ * Generate a uuid, and add it to the requestQ
+ * @param {string[]} requestQ a list of request ids
+ * @returns {uuidv4} uuid for the request
+ */
+export function generateRequestId() {
+    const requestId = uuidv4();
+    return requestId
+}
+
+/**
+ * Check the `requestId`, mark old requests as stale.
+ * Check if `requestId` is in `this.requestQ`:
+ * - IF yes: not stale, remove `requestId`, and all ids before it, return `false`
+ * - IF no: stale request, return `true`
+ * @param {string[]} requestQ a list of request ids
+ * @param {str} requestId unique identifier of request
+ * @returns {boolean, number} whether or not the request is accepted, the number of bypassed requests (excluding this request)
+ */
+export function isRequestIdStale(requestId, requestQ) {
+    const index = requestQ.indexOf(requestId, requestQ);
+
+    if (index === -1) {
+        // Not found = stale or unknown response
+        return { stale: true, staleCount: 0};
+    }
+
+    // Remove all request IDs up to and including this one
+    requestQ.splice(0, index + 1);
```

Comment:
I think this one deserves more testing. You're currently testing at most 2 pending requests. Could you test what happens if you have 5 pending requests, and the responses arrive in some order like 3 -> 5 -> 1 ->2 -> 4.

After all, mobx spine is important